src/core/xfa/factory.js

```diff
@@ -106,6 +113,45 @@ class XFAFactory {
     }
     return Object.values(data).join("");
   }
+
+  static getRichTextAsHtml(rc) {
+    if (!rc || typeof rc !== "string") {
+      return null;
+    }
+
+    try {
+      let root = new XFAParser(XhtmlNamespace, /* richText */ true).parse(rc);
+      if (!["body", "xhtml"].includes(root[$nodeName])) {
+        // No body, so create one.
+        const newRoot = XhtmlNamespace.body({});
+        newRoot[$appendChild](root);
+        root = newRoot;
+      }
+
+      const result = root[$toHTML]();
+      if (!result.success) {
+        return null;
+      }
+
+      const { html } = result;
+      const { attributes } = html;
+      if (attributes) {
+        if (attributes.class) {
+          attributes.class = attributes.class.filter(
+            attr => !attr.startsWith("xfa")
+          );
+        }
+        attributes.dir = "auto";
+      }
+
+      const text = root[$text]();
```

Comment:
Maybe calling this `str` instead, since that'd be slightly more consistent on the API/display-layer side (see comment in the `src/display/annotation_layer.js` file).